### ISSUE TITLE
question

### ISSUE THREAD

**Issue by xubangwu** [NUST-Machine-Intelligence-Laboratory/HFAN#5]
Hello, I have a problem about the CSS module. In the 260 line of hfan_vos.py, the shape of probs is (batch_size,num_classes,h*w), why the softmax function normalizes the third dimension rather than the second dimension?

**Comment by CODE4UVOS** [comment]
Thanks for your attention.

Please excuse me for not responding sooner.

```probs = F.softmax(self.scale * probs, dim=2)``` is spatial softmax, designed to normalize the foreground/background region.

You can refer to Equation 1 of the [paper](https://openaccess.thecvf.com/content_iccv_2015/html/Walker_Dense_Optical_Flow_ICCV_2015_paper.html), which may help you.
